src/riak_kv_wm_preflist.erl

```diff
@@ -0,0 +1,229 @@
+%% --------------------------------------------------------------------------
+%%
+%% riak_kv_wm_preflist - Webmachine resource for getting bucket/key active
+%%                       preflist.
+%%
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% --------------------------------------------------------------------------
+
+%% @doc Resource for getting Riak bucket/key active preflist over HTTP.
+%%
+%% Available operations:
+%%
+%% GET /types/Type/buckets/Bucket/keys/Key/preflist
+%% GET /buckets/Bucket/keys/Key/preflist
+%%   Get information about the active preflist for a particular buckey/key
+%%   combo.
+
+-module(riak_kv_wm_preflist).
+
+%% webmachine resource exports
+-export([
+         init/1,
+         service_available/2,
+         is_authorized/2,
+         forbidden/2,
+         allowed_methods/2,
+         resource_exists/2,
+         content_types_provided/2,
+         encodings_provided/2,
+         produce_preflist_body/2,
+         malformed_request/2
+        ]).
+
+-record(ctx, {
+          api_version,  %% integer() - Determine which version of the API to use.
+          bucket_type,  %% binary() - bucket type (from uri)
+          bucket,       %% binary() - Bucket name (from uri)
+          key,          %% binary() - Key (from uri)
+          client,       %% riak_client() - the store client
+          prefix,       %% string() - prefix for resource uris
+          riak,         %% local | {node(), atom()} - params for riak client
+          method,       %% atom() - HTTP method for the request
+          security      %% security context
+         }).
+
+-type context() :: #ctx{}.
+%% @doc Annotated preflist type
+-type preflist_with_number_ann() :: [ {{non_neg_integer(), node()},
```

Comment:
@hmmr take a look at https://github.com/basho/riak_core/pull/705/files#diff-b5b475eb9ebcaac228237ca394f6aaacR49 to keep consistency, as we now export partition_id.